scripts/o3de/o3de/android.py

```diff
@@ -0,0 +1,608 @@
+
+    android_configure_subparser.add_argument('--global', default=False, action='store_true',
+                                             help='Used with the configure command, specify whether the settings are project local or global.')
+    android_configure_subparser.add_argument('-p', '--project', type=str, required=False,
+                                             help="The name of the registered project to configure the settings for. This value is ignored if '--global' was specified."
+                                                  "Note: If both '--global' and '-p/--project' is not specified, the script will attempt to deduce the project from the "
+                                                  "current working directory if possible")
+    android_configure_subparser.add_argument('-l', '--list', default=False, action='store_true',
+                                             help='Display the current Android settings. ')
+    android_configure_subparser.add_argument('--validate', default=False, action='store_true',
+                                             help='Validate the settings and values in the Android settings. ')
+    android_configure_subparser.add_argument('--set-value', type=str, required=False,
+                                             help='Set the value for an Android setting, using the format <argument>=<value>. For example: \'ndk_version=22.5*\'',
+                                             metavar='VALUE')
+    android_configure_subparser.add_argument('--clear-value', type=str, required=False,
+                                             help='Clear a previously configured setting.',
+                                             metavar='VALUE')
+    android_configure_subparser.add_argument('--set-password', type=str, required=False,
+                                             help='Set the password for a password setting. A password prompt will be presented.',
+                                             metavar='SETTING')
+    android_configure_subparser.add_argument('--debug',
+                                             help=f"Enable debug level logging for this script.",
+                                             action='store_true')
+    android_configure_subparser.set_defaults(func=configure_android_options)
+
+    #
+    # Configure the subparser for 'android_generate'
+    #
+    android_generate_subparser = subparsers.add_parser(O3DE_COMMAND_GENERATE,
+                                                       help='Generate an Android/Gradle project.')
+
+    # Project Name
+    android_generate_subparser.add_argument('-p', '--project', type=str,
+                                             help="The name of the registered project to generate the Android project for. If not supplied, this operation will attempt to "
+                                                  "resolve the project from the current directory.")
```

Comment:
ditto here, in the future maybe project-path would be better? we could pull the info needed out of `project.json` if we need to get the name, version etc.